### PR TITLE
Martin/ios 645 message read in threads

### PR DESCRIPTION
**Pull request by MartinJibber**
https://user-images.githubusercontent.com/88675954/165375940-4efc346a-fa15-449a-b401-106f11e1c2db.mp4

### CODE REVIEW

File: iOS/Flows/Conversation/Cells/MessageCell.swift [hunk 1]
```diff
@@ -22,6 +22,8 @@ class MessageCell: UICollectionViewCell {
 
     let content = MessageContentView()
     private var footerView = MessageFooterView()
+    /// A view to tint the whole cell, indicating that it is unread.
+    private var unreadOverlay = BaseView()
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/MessageCell.swift:26]: The unread overlay is just a simple view with a color and transparency.

File: iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift [hunk 2]
```diff
@@ -13,7 +13,7 @@ import StreamChat
 
 class UnreadMessagesCounter: BaseView {
     
-    let imageView = UIImageView(image: UIImage(systemName: "chevron.up"))
+    let imageView = UIImageView(image: UIImage(systemName: "eyeglasses"))
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift:16]: Changing it to glasses.

File: iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift [hunk 3]
```diff
@@ -94,24 +79,31 @@ class UnreadMessagesCounter: BaseView {
         self.imageView.centerOnXAndY()
         
         self.counter.sizeToFit()
-        
-        self.countCircle.squaredSize = 20
+
+        self.countCircle.width = self.counter.width + Theme.ContentOffset.long.value
+        self.countCircle.height = 20
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift:84]: Expanding the counter to fit two digit numbers.

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 4]
```diff
@@ -220,7 +220,12 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
         let task = Task {
             guard let conversationController = self.conversationController else { return }
 
-            try? await conversationController.loadNextMessages(including: messageId)
+            // Load the message if necessary
+            if !conversationController.messages.contains(where: { message in
+                message.id == messageId
+            }) {
+                try? await conversationController.loadNextMessages(including: messageId)
+            }
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift:228]: No need to load messages if the message is already loaded. This makes scrolling to the unread message snappier.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**